subsys/dfu/dfu_target/src/dfu_stream_flatten.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <dfu_stream_flatten.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(dfu_stream_flatten, CONFIG_DFU_TARGET_LOG_LEVEL);
+
+int stream_flash_flatten_page(struct stream_flash_ctx *ctx, off_t off)
+{
+	int rc;
+	struct flash_pages_info page;
+
+	rc = flash_get_page_info_by_offs(ctx->fdev, off, &page);
+
+	if (rc != 0) {
+		LOG_ERR("Error %d while getting page info", rc);
+		return rc;
+	}
+
+	LOG_DBG("Flattening page at offset 0x%08lx", (long)page.start_offset);
+
+	rc = flash_flatten(ctx->fdev, page.start_offset, page.size);
```

Comment:
It is safe to erase "emulated page size" today (I was thinking about flattening only what is needed (the image trailer flags and MAGIC), but neverminded)
This will work today as we have proper definition of that size.